src/CoffeeShop/CoffeeShopWithDistanceToUser.php

```diff
@@ -0,0 +1,48 @@
+<?php
+namespace CoffeeShop;
+use base\ILocation;
+
+class CoffeeShopWithDistanceToUser extends CoffeeShop{
+    private $__distance;
+    private $__userLocation;
+
+    public function __construct($xCoord, $yCoord, $shopName)
+    {
+        parent::__construct($xCoord, $yCoord, $shopName);
+        $this->__userLocation = NULL;
+    }
+
+    private function setDistance($distance){
+        if(is_float($distance)){
+            $this->__distance = $distance;
+        }
+        return $this;
+    }
+
+    public function getDistance() : float{
```

Comment:
A getter that has side effects, usually not a good idea?

Reply:
This could be like a property from other languages?